app/controllers/preupgrade_reports_controller.rb

```diff
@@ -0,0 +1,8 @@
+class PreupgradeReportsController < ::Api::V2::BaseController
+  include Utils
+
+  def index
+    @preupgrade_reports = resource_scope.search_for(*search_options)
+    render :json => @preupgrade_reports.map { |report| full_report(report) }
```

Comment:
Shouldn't this reuse index layout instead of rendering json directly?

Reply:
It also probably should use rabl views for rendering json.

Reply:
That would be nice